kernelArgs_test.go

```diff
@@ -0,0 +1,48 @@
+// Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package firecracker
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestKernelArgsSerder(t *testing.T) {
+	fooVal := "bar"
+	booVal := "far"
+	emptyVal := ""
+
+	argsString := fmt.Sprintf("foo=%s blah huh=%s bleh duh=%s boo=%s",
+		fooVal,
+		emptyVal,
+		emptyVal,
+		booVal)
+
+	expectedParsedArgs := kernelArgs(map[string]*string{
+		"foo":  &fooVal,
+		"blah": nil,
+		"huh":  &emptyVal,
+		"bleh": nil,
+		"duh":  &emptyVal,
+		"boo":  &booVal,
+	})
+
+	actualParsedArgs := parseKernelArgs(argsString)
+	require.Equal(t, expectedParsedArgs, actualParsedArgs)
```

Comment:
Can you add a comment at the end like this?

```go
require.Equal(t, expectedParsedArgs, actualParsedArgs, "arguments parsed correctly")
```

Adding different comments for different `require`/`assert` calls will make it easier to disambiguate which one failed.

Also, I think these should both be `assert` instead of `require`.  Then you can see both fail and see how they failed.

Reply:
Yeah I will add the comments to the assertions.

I made this require because if the assertion fails the second assertion is going to be using input that just failed a test, so running the second assertion doesn't seem to provide much value in that case.

Reply:
You might be able to reduce dev-test cycles by having both failures shown, but I don't have strong opinions about this.